[BUILD] Windows CI vs LLVM

I'm a little stuck and could use some help/advice from somebody who is more Windows adept.

Several days back, osl's CI started failing its Windows jobs. [Here is an example log.](https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/actions/runs/15952202367/job/44993740692#step:9:742) that has this lovely error:

```
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\algorithm:8:
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\yvals_core.h(908,1): error GC074E59A: static assertion failed: error STL1000: Unexpected compiler version, expected Clang 19.0.0 or newer. [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\testshade\libtestshade.vcxproj]
    908 | _EMIT_STL_ERROR(STL1000, "Unexpected compiler version, expected Clang 19.0.0 or newer.");
        | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\yvals_core.h:534:58: note: expanded from macro '_EMIT_STL_ERROR'
    534 | #define _EMIT_STL_ERROR(NUMBER, MESSAGE)   static_assert(false, "error " #NUMBER ": " MESSAGE)
        |                                                          ^~~~~
  1 error generated.
```

Nothing changed on our end. But examining the logs, I see that on the day this started happening, the GitHub Windows runners upgraded to a new version of MSVS!

I am inferring that this newer MSVS comes with an STL that for some reason is not compatible with LLVM < 19.0 (we were, in fact, using 18.1.8). 

So I tried simply bumping the Windows jobs to use a newer LLVM, something in the 19 series (I've tried a few, but 19.1.5 is what I'm trying in the next thing I'm about to quote). That seems to make this error go away, but now I'm getting a [new link error](https://github.com/lgritz/OpenShadingLanguage/actions/runs/15957498940/job/45005724825#step:9:543) that, not being a Windows person, I'm not quite sure how to address:

```
clangFrontend.lib(LayoutOverrideSource.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
clangFrontend.lib(PrintPreprocessedOutput.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
clangFrontend.lib(CompilerInstance.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
clangFrontend.lib(TextDiagnosticPrinter.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
clangFrontend.lib(CompilerInvocation.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
clangFrontend.lib(VerifyDiagnosticConsumer.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in ast.obj [D:\a\OpenShadingLanguage\OpenShadingLanguage\build\src\liboslcomp\oslcomp.vcxproj]
...
```

This is a bit out of my depth. Can somebody with Windows experience please give me some advice, or even take a stab at fixing this yourself? I'm not sure what to try next.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -608,10 +608,10 @@ jobs:
             openexr_ver: v3.3.2
             openimageio_ver: release
             skip_tests: 1
-            setenvs: export LLVM_VERSION=18.1.8
-                            OSL_CMAKE_FLAGS="-DUSE_LLVM_BTCODE=ON"
+            setenvs: export OSL_CMAKE_FLAGS="-DUSE_LLVM_BTCODE=ON"
                             PUGIXML_VERSION=v1.14
                             OpenImageIO_BUILD_MISSING_DEPS="Freetype;TIFF;libdeflate;libjpeg-turbo"
+                            LLVM_GOOGLE_DRIVE_ID="1uy7PNVlTQ-H56unXGOS6siRWtNcdS1J7"
           - desc: Windows-2025 VS2022
             runner: windows-2025
             nametag: windows-2025
@@ -621,10 +621,10 @@ jobs:
             openexr_ver: v3.3.2
             openimageio_ver: release
             skip_tests: 1
-            setenvs: export LLVM_VERSION=18.1.8
-                            OSL_CMAKE_FLAGS="-DUSE_LLVM_BTCODE=ON"
+            setenvs: export OSL_CMAKE_FLAGS="-DUSE_LLVM_BTCODE=ON"
                             PUGIXML_VERSION=v1.14
                             OpenImageIO_BUILD_MISSING_DEPS="Freetype;TIFF;libdeflate;libjpeg-turbo"
+                            LLVM_GOOGLE_DRIVE_ID="1uy7PNVlTQ-H56unXGOS6siRWtNcdS1J7"
 
 
   optix-gpu:

--- a/src/build-scripts/gh-win-installdeps.bash
+++ b/src/build-scripts/gh-win-installdeps.bash
@@ -122,6 +122,15 @@ ls -R -l "$DEP_DIR"
 
 if [[ "$LLVM_VERSION" != "" ]] ; then
     source src/build-scripts/build_llvm.bash
+elif [[ "$LLVM_GOOGLE_DRIVE_ID" != "" ]] then
+    mkdir -p $HOME/llvm
+    pushd $HOME/llvm
+    #LLVM_GOOGLE_DRIVE_ID="1uy7PNVlTQ-H56unXGOS6siRWtNcdS1J7"
+    LLVM_ZIP_FILENAME=llvm-build.zip
+    time curl -L "https://drive.usercontent.google.com/download?id=${LLVM_GOOGLE_DRIVE_ID}&confirm=xxx" -o $LLVM_ZIP_FILENAME
+    unzip $LLVM_ZIP_FILENAME > /dev/null
+    export LLVM_ROOT=$PWD/llvm-build
+    popd
 fi
 echo "LLVM_ROOT = $LLVM_ROOT"
 

